src/sagemaker/workflow/model_step.py

```diff
@@ -282,3 +317,32 @@ def _append_repack_model_step(self):
                         "InferenceSpecification"
                     ]["Containers"][i]
                 container["ModelDataUrl"] = repacked_model_data
+
+    def _pop_out_non_configurable_repack_model_step_args(self):
+        """Pop out non-configurable args from _repack_model_step_settings"""
+        if not self._repack_model_step_settings:
+            return
+        for ignored_param in _IGNORED_REPACK_PARAM_LIST:
```

Comment:
If this is directly configurable from the user, why would they include params in their step settings that they know would get ignored ?

Reply:
The input interface to user is a dict so they can input anything they'd like. However, there are some keys/fields they should not touch, otherwise the repack model stack won't work as expected. For these field, we add this check as an early validation